test/blackbox/run_blackbox_tests/test_cases_encrypted_storage.c

```diff
@@ -0,0 +1,764 @@
+
+	meshlink_errno = MESHLINK_OK;
+	mesh = meshlink_open_encrypted("", "nut", "test", DEV_CLASS_BACKBONE, key, keylen);
+	assert_int_equal(mesh, NULL);
+	assert_int_not_equal(meshlink_errno, MESHLINK_OK);
+
+	// Opening encrypted mesh instance with NULL as key name
+
+	meshlink_errno = MESHLINK_OK;
+	mesh = meshlink_open_encrypted("encrypted_storage_conf.1", "nut", "test", DEV_CLASS_BACKBONE, NULL, keylen);
+	assert_int_equal(mesh, NULL);
+	assert_int_not_equal(meshlink_errno, MESHLINK_OK);
+
+	// Opening encrypted mesh instance with key length as 0 bytes
+
+	meshlink_errno = MESHLINK_OK;
+	mesh = meshlink_open_encrypted("encrypted_storage_conf.1", "nut", "test", DEV_CLASS_BACKBONE, key, 0);
+	assert_int_equal(mesh, NULL);
+	assert_int_not_equal(meshlink_errno, MESHLINK_OK);
+
+	return true;
+}
+
+/* Execute encrypted storage Test Case # 2 - Sanity Test */
+static void test_case_encrypted_storage_02(void **state) {
+	execute_test(test_steps_encrypted_storage_02, state);
+}
+
+/* Test Steps for encrypted storage Test Case # 2
+
+    Test Steps:
+    1. Open an encrypted confbase instance of meshlink with a key and close it.
+    2. Open the meshlink node instance with the same confbase but with a wrong or another key
+        that's different from original.
+    3. Open the meshlink node instance with the same confbase but with a valid key with which
+        initially it's opened.
+
+    Expected Result:
+    On reopening meshlink node instance with valid key @ meshlink_open_encrypted @ API should
+    return valid mesh handle else it should return NULL pointer.
+*/
+static bool test_steps_encrypted_storage_02(void) {
+	meshlink_destroy("encrypted_conf");
+	meshlink_handle_t *mesh = meshlink_open_encrypted("encrypted_conf", "nut", "test", DEV_CLASS_BACKBONE, "right", 5);
+	assert_int_not_equal(mesh, NULL);
+	meshlink_close(mesh);
+
+	mesh = meshlink_open_encrypted("encrypted_conf", "nut", "test", DEV_CLASS_BACKBONE, "wrong", 5);
+	assert_int_equal(mesh, NULL);
+
+	if(!mesh) {
+		meshlink_close(mesh);
+	}
+
+	mesh = meshlink_open_encrypted("encrypted_conf", "nut", "test", DEV_CLASS_BACKBONE, "right", 5);
+	assert_int_not_equal(mesh, NULL);
+
+	if(!mesh) {
+		meshlink_close(mesh);
+	}
+
+	meshlink_destroy("encrypted_conf");
+	return true;
+}
+
+/* Execute encrypted storage Test Case # 3 - Sanity test*/
+static void test_case_encrypted_storage_03(void **state) {
+	execute_test(test_steps_encrypted_storage_03, state);
+}
+
+/* Test Steps for encrypted storage Test Case # 3
+
+    Test Steps:
+    1. Opening meshlink node instance with confbase name that's already a directory
+        but don't contain any meshlink config files.
+
+    Expected Result:
+    @ meshlink_open_encrypted @ API should be able to open node instance with the given confbase name
+    which is already a directory.
+*/
+static bool test_steps_encrypted_storage_03(void) {
+	/*meshlink_errno = MESHLINK_OK;
+	int ret = system("rm -rf directory");
```

Comment:
You can just call `meshlink_destroy("directory")` here, instead of using `system()`.

Reply:
I think `meshlink_destroy()` only removes meshlink configuration files, but It leaves other if already there in the directory untouched right?

Reply:
Well it checks first if there's a meshlink.conf present, if not it returns an error (to prevent you from accidentily removing the wrong directory), but if it's there, it will do the equivalent of `rm -rf`, see `deltree()` in `conf.c`. But maybe it's better to keep this `system()` command then.